r1cs/gadgets/std/src/fields/nonnative/nonnative_field_gadget.rs

```diff
@@ -129,7 +129,7 @@ impl<SimulationF: PrimeField, ConstraintF: PrimeField>
     /// A function for test purposes. Returns `true` if `&self.num_add` respects 
     /// the capacity bound, and bounds all the limbs correctly.
     pub(crate) fn check(&self) -> bool {
-        let params = get_params(SimulationF::size_in_bits(), ConstraintF::size_in_bits());
```

Comment:
It can be done but this would require to expose a public function `get_params_for_pseudomersenne`, which maybe is a detail we want to hide. A solution may be exposing a non constant public function `get_params` which then relies on two const private functions for pseudo-mersenne and non pseudo-mersenne. And maybe the same for `find parameters`, which must be public in the crate to be employed in unit tests.

Reply:
Yes i agree let's proceed in this way

Reply:
Right now I implemented a public non constant `get_params` function which simply calls the specific constant functions for pseudo-mersenne or non psuedo-mersenne fields. I made these two specific functions `get_params_for_pseudomersenne` and `get_params_for_generic_field` public in the crate in order to be directly called from the `NonNativeFieldGadget` code whenever it is already known that the simulation field is pseudo-mersenne or not.
In case we prefer to keep these two functions private, we can just replace them with the non-constant `get_params` function.